### PR TITLE
Tooltip: Have `preferBelow` default to false instead of true

### LINKED ISSUE #133006: Tooltip: Have `preferBelow` default to false instead of true
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

(All use cases for the Tooltip widget.)

### Proposal

[`Tooltip.preferBelow`](https://api.flutter.dev/flutter/material/Tooltip/preferBelow.html) defaults to `true`, but when the tooltip appears below instead of above, it's likely to be partly hidden under a user's finger or cursor. So I think `false` is a better default.

The [Material Design 3 doc](https://m3.material.io/components/tooltips/guidelines#57286ec2-db8e-4699-aa43-0ca11cab0bb5) agrees:

> By default, plain tooltips are positioned directly above the parent element.

### PR DESCRIPTION
**Pull request by chrisbobbe**
Fixes #133006.

Also, update the dartdoc on `preferBelow` to mention the existing role of [TooltipThemeData.preferBelow].

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by TahaTesser — APPROVED** [review]
LGTM!

**Comment by TahaTesser** [comment]
We'll need a second review before landing this.

**Comment by HansMuller** [comment]
This is a sensible change but it's also a breaking change - the "Google Testing" failures are evidence of that.  I think we'd be better off with an example that showed how to configure an app's TooltipTheme to do the right thing by default

**Comment by chrisbobbe** [comment]
Makes sense. I plan to send that docs change and close #133006 as WONTFIX, then.

**Pull request closed without merging**
